server/controllers/api/oEmbed/getOEmbedDataForAsset.js

```diff
@@ -0,0 +1,60 @@
+const logger = require('winston');
+const db = require('../../../models');
+const getClaimId = require('../../utils/getClaimId');
+
+const getOEmbedDataForAsset = (channelName, channelClaimId, claimName, claimId) => {
+  let fileData, claimData;
+  let data = {
+    version      : '1.0',
+    provider_name: 'Spee.ch',
+    provider_url : 'https://spee.ch',
+    cache_age    : 86400, // one day in seconds
+  };
+
+  return getClaimId(channelName, channelClaimId, claimName, claimId)
+    .then(fullClaimId => {
+      claimId = fullClaimId;
+      return db.Claim.findOne({
+        where: {
+          name   : claimName,
+          claimId: fullClaimId,
+        },
+      });
+    })
+    .then(claimRecord => {
+      claimData = claimRecord.dataValues;
+      return db.Blocked.isNotBlocked(claimData.outpoint);
+    })
+    .then(() => {
+      return db.File.findOne({
+        where: {
+          name: claimName,
+          claimId,
+        },
+      });
+    })
+    .then(fileRecord => {
+      fileData = fileRecord.dataValues;
+      logger.debug('file data:', fileData);
+      const serveUrl = `https://dev1.spee.ch/${fileData.claimId}/${fileData.name}.${fileData.fileType.substring(fileData.fileType.indexOf('/') + 1)}`;
```

Comment:
Do you have configs for the host URL generation?
That would be ideal over hardcoding

Reply:
good catch.  I meant to do this after testing and forgot.